DM-51011: Add a X-Kernel-Name request header to /execution

jonathansick opened the pull request:
This provides a way to specify what kernel the notebook should be executed with, with a fallback to the default kernel or the kernel in the notebook metadata (current behaviour).

jonathansick committed:
Add a X-Kernel-Name request header to /execution

This provides a way to specify what kernel the notebook should be
executed with.

jonathansick reviewed (commented):
I've tested this with the `us-central1-docker.pkg.dev/rubin-shared-services-71ec/sciplat/sciplat-lab:exp_w_2025_20_execution-kernel@sha256:c279cd7332de7a5a572d1a0cd6c54820a0ac9bac17f720f026cbe60c9d3c06f3` image build and it does work. A notebook with a `python3` kernel spec built-in was executed with the `lsst` kernel by providing the `X-Kernel-Name` header.

athornton commented on rsp_jupyter_extensions/handlers/execution.py:
```diff
@@ -20,25 +20,32 @@ def rubinexecution(self) -> dict[str, str]:
+        execution.
         """
         input_str = self.request.body.decode("utf-8")
+        kernel_name = self.request.headers.get("X-Kernel-Name", None)
```
I think `None` is a better choice than `"LSST"` because this stuff goes in the base container, which may be used to construct images with non-DM-stack payloads, and those would not even have an `LSST` kernel.

But I had to think about it for a few minutes to make up my mind.

jonathansick replied:
Keeping None as the default maintains the current behaviour, so I think it makes sense.

athornton reviewed (approved):
LGTM, and thank you for improving the documentation.

jonathansick committed:
Add tests for execution extension

These tests are mock-heavy, but ensure that the kernel_name is being
treated correctly.
